extensions/default/src/DicomJSONDataSource/index.js

```diff
@@ -258,19 +262,19 @@ function createDicomJSONApi(dicomJsonConfig) {
 
       const { StudyInstanceUID, SeriesInstanceUID } = displaySet;
       const study = findStudies('StudyInstanceUID', StudyInstanceUID)[0];
-      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID);
-      let instances = displaySet.images;
-      if (series.instances.length > displaySet.images.length) {
-        instances = series.instances;
-      }
+      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID) || [];
 
-      instances.forEach(instance => {
+      displaySet.images.forEach(instance => {
         const NumberOfFrames = instance.NumberOfFrames;
-
         if (NumberOfFrames > 1) {
           for (let i = 0; i < NumberOfFrames; i++) {
```

Comment:
Can you re-implement this to deal with large number of frames and/or large number of sop instances in a series?  THe JSON data should be able to just reference an instance with NumberOfFrames in it, and the URL not having a per-frame value in it, and be able to find that easily.
That is, allowing either the representation where every frame is present in the JSON or just a single multiframe is present in the JSON should both be allowed, and should match up correctly, without needing to go over every single frame and match up to every single image instance by looking at every JSON image instance for each frame.

Reply:
This implementation should work for both. This was a change for an old project, do you think its a quick change to deal with large number of frames?

Reply:
Updated!